src/xbot/common/logging/PeriodicDeltaObserver.java

```diff
@@ -0,0 +1,30 @@
+package xbot.common.logging;
+
+import org.apache.log4j.Logger;
+
+public class PeriodicDeltaObserver {
+
+    protected static Logger log = Logger.getLogger(PeriodicDeltaObserver.class);
+    
+    private double checkPowerThreshold;
+    
+    private double oldPower = 0;
+    
+    public PeriodicDeltaObserver(double checkPowerThreshold) {
+        this.checkPowerThreshold = checkPowerThreshold;
+    }
+    
+    public void setCheckPowerThreshold(double power) {
+        this.checkPowerThreshold = power;
+    }
+    
+    public boolean isThereASignficantChangeInTheSystem(double newPower) {
+        if (Math.abs(oldPower - newPower) > checkPowerThreshold) {
+            oldPower = newPower;
+            return true;
+        }
+        oldPower = newPower;
```

Comment:
If we didn't notify, we don't want to replace the power value. We want to notify every time the value has changed by more than the threshold _since the last time the caller was notified_.